Riot/Modules/MatrixKit/Models/Account/MXKAccountData.m

```diff
@@ -80,6 +80,15 @@ - (id)initWithCoder:(NSCoder *)coder
 
         _warnedAboutEncryption = [coder decodeBoolForKey:@"warnedAboutEncryption"];
         
+        if ([coder decodeIntForKey:@"preferredSyncPresence"])
```

Comment:
This check will still return false even if `preferredSyncPresence` is `MXPresenceUnknown`. So it depends on the value of presence enum. It makes sense to encode/decode string value of the presence instead.